lib/ansible/galaxy/role.py

```diff
@@ -418,13 +425,26 @@ def install(self):
                                 relative_path = os.path.join(*full_path.replace(relative_path_dir, "", 1).split(os.sep))
                                 setattr(member, attr, relative_path)
 
+                            if member in ignore_external:
+                                continue
+
                             if _check_working_data_filter():
                                 # deprecated: description='extract fallback without filter' python_version='3.11'
-                                role_tar_file.extract(member, to_native(self.path), filter='data')  # type: ignore[call-arg]
+                                role_tar_file.extract(member, temporary_dest, filter='data')  # type: ignore[call-arg]
                             else:
                                 # Remove along with manual path filter once Python 3.12 is minimum supported version
-                                role_tar_file.extract(member, to_native(self.path))
+                                role_tar_file.extract(member, temporary_dest)
+
+                        extract_complete = True
+
+                        if os.path.exists(self.path) and not self.remove():
+                            raise AnsibleError("%s doesn't appear to contain a role.\n  please remove this directory manually if you really "
+                                               "want to put the role here." % self.path)
+                        if not os.path.exists(self.path):
+                            os.makedirs(self.path)
 
+                        for path in os.listdir(temporary_dest):
```

Comment:
Wouldn't it be more atomic to move the entire directory rather than parts of it one by one?

Reply:
This really needs tests first, please disregard the draft.

Reply:
I was just rebasing PRs with the stale_pr label.